seamapi/python-legacy
# Introduce Generated SDK

- generated sdk replaces old sdk
- initial test workspace test working
- validate workspaces methods - one missing
- thermostats climate setting schedules blocked by no 'climate_setting_schedules
- Revert "validate workspaces methods - one missing"
- remove factory
- revert thermostat changes
- add generate in development mode
- remove publish and version workflows (for now)
- update generated SDK
- update python SDK output
- update SDK output and test_workspaces
- wip: updating tests
- thermostat climate setting schedules are passing - check for prop er usage of ids
- working access_codes, wip: SeamApiException
- SeamApiException fix
- connected_accounts and events exceptions tests pass
- add preferred_code_length tests
- add .gitattributes
- working noise_sensors/noise_thresholds
- workspaces test passing, update sentry errors test
- update access codes to accomodate return None
- noise sensors test passing
- update test for None return
- change accesing properties to dot notation
- update action attempts test to work on unlock
- update generated SDK code
- poll_action_attempt implement, Union type import
- update generation output
- update output
- add dependabot to update all @seamapi dependencies
- add auto-merge workflow
- specify what to test in TODO
- add .nvmrc
- test dependabot
- add dependency type
- more dependabot tests
- specify npm registry for @seamapi dependencies
- update to @seamapi deps to use npm registry
- change auto merge pattern
- working dependabot auto merge workflow
- chore(deps): bump the seam group with 1 update
- chore(deps): bump the seam group with 1 update
- Setup generation workflow
- Update nextlove-sdk-generator registry in package lock
- Update package lock
- Update generate workflow
- Add format step to generate workflow
- ci: Generate code
- chore(deps-dev): bump the seam group with 2 updates
- Update generate workflow to run when a PR opens or new commits pushed to it
- Revert "Add `pull_request_target` event to generate workflow"
- ci: Generate code
- chore(deps-dev): bump the seam group with 2 updates
- ci: Generate code
- update nextlove and output
- auto approve dependabot PRs
- ci: Generate code
- chore(deps-dev): bump the seam group with 1 update
- chore(deps-dev): bump the seam group with 1 update
- Generate script should be terminated if any command exits with a non-zero status
- ci: Generate code
